### DISCUSSION TITLE
Add ability to access to the httpRequest headers via Session

### DISCUSSION THREAD

**Discussion by damienissa** [serverpod/serverpod#831] [Q&A] [answered]
Add the ability to access and manipulate HTTP headers in requests sent to the server. Serverpod provides a convenient way to handle HTTP requests and responses within a Flutter app. Still, it would be helpful if the server could handle requests from other sources, such as websites or APIs.

Overall, adding the ability to access and manipulate HTTP headers in Serverpod would significantly enhance its flexibility and usefulness for a broader range of applications.

**Comment by bk-one** [comment]
In general we just need a way to validate an incoming request. Our initial idea was to just send a custom http header and validate the header. Alternatively just allowing a standard `Authorization: Bearer xxx` header that we can validate ourselves. In any case, being able to allow incoming external requests to handle incoming webhooks is what we need.

**Reply by vlidholt** [reply]
Have you looked at the auth module?
https://docs.serverpod.dev/concepts/authentication

**Comment by vlidholt** [comment, marked as answer]
This feature already exists. :) For example, to access the IP number of the caller, you can use:

```yaml
session as MethodCallSession;
var ipAddress = session.httpRequest.remoteIpAddress;
```

If you are authenticating a user, use the auth module and Serverpod will handle everything related to keeping track of users tokens etc.

An option for communication with other servers, e.g. using webhooks, is to use Serverpod's web server, Relic. Some docs here:
https://docs.serverpod.dev/concepts/webserver

**Comment by perumalrollfi** [comment]
Hi Viktor,

Tried using the above solution. But getting ip address as below.
  ::e8a2:a797:f77f:0%2546119632

is there any other setting needed to get ipaddress in correct form?

Thanks
Perumal

**Reply by vlidholt** [reply]
What platform are you running Serverpod on?

**Reply by perumalrollfi** [reply]
Hi Viktor, 

We are running serverpod on AWS EC2 Amazon Linux distribution.

Thanks
Perumal

**Comment by yahu1031** [comment]
Currently @damienissa @bk-one @perumalrollfi I think this will help you out regarding accessing headers from the session in the backend(server) https://github.com/serverpod/serverpod/issues/663#issuecomment-1500974669

**Comment by dunatron** [comment]
I have a use case where my application is going to be multi tenant.

A user can have multiple companies. Most requests will have a. companyid attached to the header.
I would like a token also so i can actually validate they have access to said company.

**Reply by dunatron** [reply]
Found it i guess. your AI is not aware of it 
https://docs.serverpod.dev/concepts/authentication/custom-overrides
